src/main/java/org/apache/mesos/offer/OfferEvaluator.java

```diff
@@ -310,6 +331,10 @@ private TaskInfo getFulfilledTaskInfo(
               ExecutorInfo.newBuilder(execInfo)
                       .clearResources();
 
+      if (execInfo.getExecutorId().getValue().isEmpty()) {
+        execBuilder.setExecutorId(ExecutorUtils.toExecutorId(execInfo.getName()));
+      }
+
```

Comment:
This isn't the right place for this.  I'm going to move it up the ExecutorReqirement fulfillment section.